backend/timed/tracking/tests/test_attendance.py

```diff
@@ -127,3 +127,24 @@ def test_attendance_to_before_from(internal_employee_client, attendance_factory)
     assert json["errors"][0]["detail"] == (
         "An attendance may not end before it starts."
     )
+
+
+def test_attendance_can_end_at_midnight(internal_employee_client, attendance_factory):
```

Comment:
Might be worth it to use `@pytest.mark.parametrize` on the test above and test a full set of edge cases. Something like

```python
@pytest.mark.parametrize(
    "attendance__from_time, attendance__to_time, set_from_time, set_to_time expect_result",
    [
        (time(7,30), time(8,30), "7:30", "00:00", status.HTTP_200_OK),
        (time(7,30), time(8,30), "7:00", "00:00", status.HTTP_400_BAD_REQUEST),
        ...
    ]
)
def test_attendance_modification(attendance, set_to_time, set_from_time, internal_employee_client):
    ...
```